qemu/qmp.go

```diff
@@ -828,3 +829,53 @@ func (q *QMP) ExecuteQueryHotpluggableCPUs(ctx context.Context) ([]HotpluggableC
 
 	return cpus, nil
 }
+
+// ExecSetMigrationCaps sets migration capabilities
+func (q *QMP) ExecSetMigrationCaps(ctx context.Context, caps []map[string]interface{}) error {
+	args := map[string]interface{}{
+		"capabilities": caps,
+	}
+
+	return q.executeCommand(ctx, "migrate-set-capabilities", args, nil)
+}
+
+// ExecSetMigrateArguments sets the command line used for migration
+func (q *QMP) ExecSetMigrateArguments(ctx context.Context, url string) error {
+	args := map[string]interface{}{
+		"uri": url,
+	}
+
+	return q.executeCommand(ctx, "migrate", args, nil)
+}
+
+// ExecHotplugMemory adds size of MB memory at specified slot to the guest
+func (q *QMP) ExecHotplugMemory(ctx context.Context, slot, size int) error {
+	id := "mem" + strconv.Itoa(slot)
+	args := map[string]interface{}{
+		"qom-type": "memory-backend-ram",
+		"id":       id,
+		"props":    map[string]interface{}{"size": int64(size) << 20},
+	}
+	err := q.executeCommand(ctx, "object-add", args, nil)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			err = q.executeCommand(ctx, "object-del", map[string]interface{}{"id": id}, nil)
```

Comment:
If this commands fails we will lost the first error , could you log it before call `object-del`?

Reply:
Will do.